filebeat/registrar/registrar.go

```diff
@@ -176,6 +168,32 @@ func (r *Registrar) loadAndConvertOldState(f *os.File) bool {
 	return true
 }
 
+func convertOldStates(oldStates map[string]file.State) []file.State {
```

Comment:
Hmm, that would simpler and would work, but bypassing the `IsSame()` method feels risky for future `StateOS` implementations (note that the structures is platform specific, so we could see variations).

I'm not really worried about the O(n^2) aspect here, since this is done only once and with an N up to tens of thousands at most.